packages/expo-mail-composer/android/src/main/java/expo/modules/mailcomposer/MailIntentBuilder.kt

```diff
@@ -0,0 +1,83 @@
+package expo.modules.mailcomposer
+
+import android.app.Application
+import android.content.ComponentName
+import android.content.Intent
+import android.net.Uri
+import android.text.Html
+import android.util.Log
+import androidx.core.content.FileProvider
+import expo.modules.core.arguments.ReadableArguments
+import java.io.File
+
+class MailIntentBuilder(
+  private val options: ReadableArguments
+) {
+  private val mailIntent = Intent(Intent.ACTION_SEND_MULTIPLE)
+
+  private fun getStringArrayFrom(key: String): Array<String?> {
+    return (options.getList(key) as List<String?>).toTypedArray()
+  }
+
+  private fun contentUriFromFile(file: File, application: Application): Uri = try {
+    FileProvider.getUriForFile(
+      application,
+      application.packageName + ".MailComposerFileProvider",
+      file
+    )
+  } catch (e: Exception) {
+    Uri.fromFile(file)
+  }
+
+  fun build() = mailIntent
+
+  fun setComponentName(pkg: String, cls: String): MailIntentBuilder {
+    mailIntent.component = ComponentName(pkg, cls)
+    return this
+  }
```

Comment:
```suggestion
  fun setComponentName(pkg: String, cls: String): MailIntentBuilder = run {
    mailIntent.component = ComponentName(pkg, cls)
  }
```

Reply:
I suppose you meant `apply` instead of `run` ;)